### PR TITLE
dhclient: Poll until IPv6 address is not tentative

### PR DESCRIPTION
**Pull request by rjoleary**
Signed-off-by: Ryan O'Leary <ryanoleary@google.com>

### CODE REVIEW

**Comment by rjoleary** [comment]
Still testing...

File: pkg/dhclient/dhclient.go [hunk 1]
```diff
@@ -40,6 +59,15 @@ func IfUp(ifname string) (netlink.Link, error) {
 		}
 
 		if iface.Attrs().Flags&net.FlagUp == net.FlagUp {
+			// For ipv6, we cannot bind to the port until Duplicate
+			// Address Detection (DAD) is complete which is
+			// indicated by the link being no longer marked as
+			// "tentative".
+			if ipv6 && !isIpv6LinkReady(ifname, iface) {
```
Comment (rjoleary) [inline pkg/dhclient/dhclient.go]: This doesn't work so well when the interface has available ipv4 addresses, but not ipv6 addresses.

File: pkg/dhclient/dhclient.go [hunk 2]
```diff
@@ -126,6 +146,22 @@ func lease4(ctx context.Context, iface netlink.Link, timeout time.Duration, retr
...
+	// Detection (DAD) is complete which is indicated by the link being no
+	// longer marked as "tentative". This usually takes about a second.
+	for {
+		if ready, err := isIpv6LinkReady(iface); err != nil {
+			return nil, err
+		} else if ready {
+			break
+		}
+		select {
+		case <-time.After(100 * time.Millisecond):
```
Comment (hugelgupf) [inline pkg/dhclient/dhclient.go]: is 100 ms enough? maybe 500?

File: pkg/dhclient/dhclient.go [hunk 3]
```diff
@@ -126,6 +146,22 @@ func lease4(ctx context.Context, iface netlink.Link, timeout time.Duration, retr
 }
 
 func lease6(ctx context.Context, iface netlink.Link, timeout time.Duration, retries int) (Lease, error) {
+	// For ipv6, we cannot bind to the port until Duplicate Address
+	// Detection (DAD) is complete which is indicated by the link being no
+	// longer marked as "tentative". This usually takes about a second.
```
Comment (hugelgupf) [inline pkg/dhclient/dhclient.go]: "usually takes about a second" conflicts with time.After 100ms

File: pkg/dhclient/dhclient.go (hunk 2 above)
Comment (hugelgupf) [inline pkg/dhclient/dhclient.go]: did you mean for this to have a `fallthrough`?

Comment (rjoleary) [reply pkg/dhclient/dhclient.go]: Added continue to make it more clear.

**Pull request merged**